pkg/webservice/dao/scheduler-health.go

```diff
@@ -0,0 +1,37 @@
+/*
+ Licensed to the Apache Software Foundation (ASF) under one
+ or more contributor license agreements.  See the NOTICE file
+ distributed with this work for additional information
+ regarding copyright ownership.  The ASF licenses this file
+ to you under the Apache License, Version 2.0 (the
+ "License"); you may not use this file except in compliance
+ with the License.  You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package dao
+
+type SchedulerHealthDAOInfo struct {
+	Healthy              bool                         `json:"healthy"`
+	SchedulingErrors     int                          `json:"schedulingErrors"`
+	FailedNodes          int                          `json:"failedNodes"`
+	UnhealthyPartition   map[string]*PartitionDAOInfo `json:"unhealthyPartitions"`
+	UnhealthyNodes       map[string]*NodeDAOInfo      `json:"unhealthyNodes"`
+	ReservationNodeRatio float32                      `json:"reservationRatio"`
+}
+
+func (s *SchedulerHealthDAOInfo) SetHealthStatus() {
+	s.Healthy =
+		s.SchedulingErrors == 0 &&
+			s.FailedNodes == 0 &&
+			len(s.UnhealthyPartition) == 0 &&
+			len(s.UnhealthyNodes) == 0
+	//TODO: what is a healthy reservation/node ratio?
```

Comment:
if we do not know what's the reasonable ratio, can we add this value to the returned message and mark this check always passed? So we can observe what kind of values we can get from the real cluster. maybe later we have a better idea of how to set this up.